stable_diffusion_videos/stable_diffusion_walk.py

```diff
@@ -192,7 +204,11 @@ def walk(
                 embeds = slerp(float(t), embeds_a, embeds_b)
             latents = slerp(float(t), latents_a, latents_b)
 
-            with torch.autocast("cuda"):
+            device = choose_torch_device()
+            precision_scope = torch.autocast
+            if device in ['mps', 'cpu']:
```

Comment:
really happy you patched this to work on CPU if needed (though it would be terrifyingly slow lol).